src/main/java/org/folio/rest/impl/EholdingsProvidersImpl.java

```diff
@@ -180,8 +191,45 @@ public void putEholdingsProvidersByProviderId(String providerId, String contentT
 
 
   @Override
+  @Validate
   @HandleValidationErrors
-  public void getEholdingsProvidersPackages(String q, String filterSelected, String filterType, String sort, int page, int count, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(Future.succeededFuture(GetEholdingsProvidersResponse.status(Response.Status.NOT_IMPLEMENTED).build()));
+  public void getEholdingsProvidersPackagesByProviderId(String providerId, String q, String filterSelected,
+                                                        String filterType, String sort, int page, int count,
+                                                        Map<String, String> okapiHeaders,
+                                                        Handler<AsyncResult<Response>> asyncResultHandler,
+                                                        Context vertxContext) {
+    long providerIdLong;
```

Comment:
In all other cases we validate that q is not empty and return an error if it's empty, I think it's better to do it here too, for consistency

Reply:
Thanks, updated